### PR TITLE
Expanded test coverage of django.utils.dateformat

### PR DESCRIPTION
**Pull request by ngnpope**
Split out from #15405 according to the plan in https://github.com/django/django/pull/15405#issuecomment-1295325606.

### CODE REVIEW

File: tests/utils_tests/test_dateformat.py [hunk 1]
```diff
@@ -142,9 +142,23 @@ def test_time_formats(self):
...
+            # (Not using raw strings here as they cannot end with a backslash.)
+            ("\\jS \\o\\f F", "jth of July"),
+            ("jS \\o\\f F\\", "8th of July\\"),
+            # More pathological cases of nested escaping.
+            (r"Y z \C\E\T", "1979 189 CET"),
+            (r"Y z \\\C\\\E\\\T", r"1979 189 \C\E\T"),
+            (r"Y z \\\\\C\\\\\E\\\\\T", r"1979 189 \\C\\E\\T"),
+            # FIXME: These should work, but the current implementation is broken.
+            # (r"Y z \\C\\E\\T", r"1979 189 \C\July\CET"),
+            # (r"Y z \\\\C\\\\E\\\\T", r"1979 189 \\C\\July\\CET"),
```
Comment (felixxm) [inline tests/utils_tests/test_dateformat.py:158]: I would move these lines to PR with the appropriate fix. Do we have one?

Comment (ngnpope) [reply tests/utils_tests/test_dateformat.py:158]: Sure, will do. This will be fixed as part of the main PR.

Comment (felixxm) [reply tests/utils_tests/test_dateformat.py:158]: Great, thanks!

**Comment by felixxm** [comment]
@ngnpope Thanks :+1: What about the last two points from your plan?
```
...and add better/missing coverage of datetime.{date,time} here?
...and add better/missing coverage of django.utils.dateformat.time_format() here?
```

**Comment by ngnpope** [comment]
> What about the last two points from your plan?

That's why it's a draft... 😉 Will be sorting them out over the next few days. Need to refresh my memory of where I was at

**Comment by felixxm** [comment]
> > What about the last two points from your plan?
> 
> That's why it's a draft... wink Will be sorting them out over the next few days. Need to refresh my memory of where I was at

Sounds like a plan :+1: Maybe on Django Con Europe sprints :scotland:

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**